### PR TITLE
Fix Virology Healing symptom exploit.

### PR DESCRIPTION
**Pull request by ChakatStormCloud**
## About The Pull Request
This removes a Stage Speed 6 threshold that's sat in the symptom/heal class as a template for far too long, that had been silently buffing every single healing symptom in an undocumented and uninteded way. Most obvious is with the regenerative coma symptom, which has a stage speed 7 threshhold setting the power to 1.5, where the exploit, at 6 would put it at power = 2. 

EDIT: To clarify.
Deleting these 2 lines of code, in a template symptom that doesn't do anything, and says not to use it, fixes the following issues:
```
Starlight Condensation (heal/starlight): duplicated SS6:Power=2  threshold code.
Toxolysis (heal/chem):                   duplicated Power=2 at R7, undocumented SS6:Power=2 threshold
Nocturnal Regeneration (heal/darkness):  redundant SS8:Power=2 code, Pandemic showing SS8 when it happens at SS6 
Regenerative Coma (heal/coma):           undocumented SS6:Power=2 sweet spot, before the documented SS7:Power=1.5
Tissue Hydration (heal/water):           redundant SS7:Power=2 code, Pandemic showing SS7 when it happens at SS6 
Plasma Fixation (heal/plasma):           redundant SS7:Power=2 code, Pandemic showing SS7 when it happens at SS6 
Radioactive Resonance(heal/radiation):   duplicated Power=2 at R7, undocumented SS6:Power=2 threshold
*SS: StageSpeed, R7: Resistance 7
```

This happens because they all call ..() in Start(), which calls heal/Start().
This cannot be removed because heal/Start() calls ..() to call symptoms/Start() and inherit the base neutering behaviour.

## Why It's Good For The Game
Removes an ages old exploit and keeps the healing symptoms functioning as intended, and as documented.

## Changelog

:cl:
fix: Virology: all healing symptoms no longer receive an accidental power boost at Stage Speed 6.
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
https://github.com/tgstation/tgstation/blob/5d41db27c805cdcddb2825217d3affa2db4fc45c/code/datums/diseases/advance/symptoms/heal.dm#L14 It seems intended?

**Comment by Nidvex250** [comment]
> https://github.com/tgstation/tgstation/blob/5d41db27c805cdcddb2825217d3affa2db4fc45c/code/datums/diseases/advance/symptoms/heal.dm#L14
> 
> It seems intended?

hmm... it brings up a point though; you would have to dig into the code to find out about this threshold. Probably says something that Regenerative Coma actually gets worse when meeting the listed threshold of 7 instead of getting better at what it does due to this section of code existing.
The real question may be if we want Regenerative Coma's best to be 2.0 or 1.5, which will answer what lines of code we'd want to remove or change to fix this discrepancy. If nothing else this definitely seems undocumented.

**Comment by Colovorat** [comment]
i am not sure why you thinking it's an exploit. it's like intended and was in the game for ages

**Comment by Mothblocks** [comment]
@Nidvex250 I'm confused--that text shows up in the PANDemic when you reach the threshold.

This doesn't look like an exploit.

**Comment by ChakatStormCloud** [comment]
@Nidvex250 
It mentions that for the base heal, which was a template, and isn't supposed to do anything on it's own.

@Colovorat @Nidvex250 
This is absolutely an unintended exploit, look at every single symptom in Heal.dm, every single one except heal/metabolism has it's own Power =2 or Power =1.5 (in the case of heal/coma).

This effects ALL of them, not just regenerative coma.
and NO it does NOT show this in the PANDemic, it shows the Stage Speed 7 threshold, it does not mention a Stage Speed 6 threshold.

**Starlight Condensation(heal/starlight):** literally duplicating the code in it's own Start(), if this was intentional why?
**Toxolysis (heal/chem):** has it's own power = 2 at resistance 7, also doesn't mention the same happening at stagespeed 6, only mentions it's own coded threshold there.
**Metabolic Boost(heal/metabolism):** doesn't use the power variable.
**Nocturnal Regeneration(heal/darkness):** Has it's own Power =2 that doesn't trigger until stage speed EIGHT, again, because of this, it triggers at 6 instead, the PANDemic only mentions this at stage speed 8, like it's written, redundantly, in darkness/Start()
**Regenerative Coma(heal/coma):** I already mentioned this, and again no, the PANDemic shows this triggering at stagespeed 7 not 6.
**Tissue Hydration(heal/water):** same as issue as heal/darkness, it duplicates the code intending to trigger at StageSpeed 7, and again PANDemic mentions StageSpeed 7, not 6.
**Plasma Fixation(heal/plasma):** Exact same issue as heal/water
**Radioactive Resonance(heal/radiation):** Exact Same issue as heal/water again.

**Comment by ChakatStormCloud** [comment]
@Mothblocks if you still think this is intended, look at the other PR, that's the consequences of those 2 lines of code in heal/Start()

Inheritance, it's not that hard to understand.

**Comment by Mothblocks** [comment]
> @Mothblocks if you still think this is intended, look at the other PR, that's the consequences of those 2 lines of code in heal/Start()
> 
> Inheritance, it's not that hard to understand.

I don't have any obligation to help with you on this if you're just going to be rude about it.

Anyway this was a misunderstanding and ATHATH has told me this is an exploit, which is an unbelievable crown to get

**Comment by ChakatStormCloud** [comment]
> I don't have any obligation to help with you on this if you're just going to be rude about it.
> 
> Anyway this was a misunderstanding and ATHATH has told me this is an exploit, which is an unbelievable crown to get

Sorry for being an arse about it, I woke up to 4 people having skimmed the thing and closed it without actually understanding the nature of the exploit, and that's a massive pet peeve of mine. Though in fairness I did skip over all the effects of it, and only mentioned the worst effect of it, and I've learned from that and should have made it clear from the start.

This should be ready to merge as is if the map integration tests didn't get cancelled for some reason...

File: code/datums/diseases/advance/symptoms/heal.dm [hunk 1]
```diff
@@ -10,17 +10,13 @@
 	symptom_delay_min = 1
 	symptom_delay_max = 1
 	var/passive_message = "" //random message to infected but not actively healing people
-	threshold_descs = list(
-		"Stage Speed 6" = "Doubles healing speed.",
-		"Stealth 4" = "Healing will no longer be visible to onlookers.",
```
Comment (Mothblocks) [inline code/datums/diseases/advance/symptoms/heal.dm:15]: Is the stealth thing old?

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by ATH1909** [comment]
> Sorry for being an arse about it, I woke up to 4 people having skimmed the thing and closed it without actually understanding the nature of the exploit, and that's a massive pet peeve of mine.

I feel ya, man. I apologize for not chiming in earlier; I haven't been checking GitHub very often lately due to focusing on my move+new job.

File: code/datums/diseases/advance/symptoms/heal.dm (hunk 1 above)
Comment (ChakatStormCloud) [reply code/datums/diseases/advance/symptoms/heal.dm:15]: Yeah, I checked the code, there's nothing that acts to that effect, not to mention a symptom that's only meant to provide common code shouldn't have it's own thresholds

**Pull request merged**